From a4e0398eeeb1a37980a7995fcbb745adca02961d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 14 Oct 2026 00:25:25 +0200
Subject: [PATCH] desktop: Fix desktop window position with client-side
 decorations

Move the desktop window after setting its decorated property to
FALSE, so GTK can correctly account for client-side decorations
when positioning the window.

This fixes the desktop window being positioned at negative
coordinates on X11 when GTK_CSD=1 is used.
---
 src/caja-desktop-window.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/caja-desktop-window.c b/src/caja-desktop-window.c
index 8193c6507..7bf189616 100644
--- a/src/caja-desktop-window.c
+++ b/src/caja-desktop-window.c
@@ -103,8 +103,6 @@ caja_desktop_window_init (CajaDesktopWindow *window)
     context = gtk_widget_get_style_context (GTK_WIDGET (window));
     gtk_style_context_add_class (context, "caja-desktop-window");
 
-    gtk_window_move (GTK_WINDOW (window), 0, 0);
-
     /* shouldn't really be needed given our semantic type
      * of _NET_WM_TYPE_DESKTOP, but why not
      */
@@ -221,6 +219,9 @@ caja_desktop_window_new (CajaApplication *application,
                               "screen", screen,
                               "decorated", FALSE,
                               NULL));
+
+    gtk_window_move (GTK_WINDOW (window), 0, 0);
+
     /* Stop wrong desktop window size in GTK 3.20*/
     /* We don't want to set a default size, which the parent does, since this */
     /* will cause the desktop window to open at the wrong size in gtk 3.20 */
